masu/external/accounts/network/cur_accounts_network.py

```diff
@@ -0,0 +1,114 @@
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU Affero General Public License for more details.
+#
+# You should have received a copy of the GNU Affero General Public License
+# along with this program.  If not, see <https://www.gnu.org/licenses/>.
+#
+"""Database source impelmentation to provide all CUR accounts for CURAccounts access."""
+
+import requests
+
+from masu.config import Config
+from masu.external.accounts.cur_accounts_interface import (CURAccountsInterface,
+                                                           CURAccountsInterfaceError)
+
+
+# pylint: disable=too-few-public-methods
+class CURAccountsNetwork(CURAccountsInterface):
+    """Provider interface defnition."""
+
+    def __init__(self):
+        """Initializer."""
+        self.koku_host = Config.KOKU_HOST
+        self.koku_port = str(Config.KOKU_PORT)
+        self.admin_user = Config.KOKU_ADMIN_USER
+        self.admin_pass = Config.KOKU_ADMIN_PASS
+
+    def _get_service_admin_token(self):
+        """
+        Retrieve service admin token from Koku.
+
+        This will call koku's token-auth to get the service admin token.
+
+        Args:
+            None
+
+        Returns:
+            (String) : Service Admin token
+
+        """
+        token_auth_url = 'http://{}:{}/api/v1/token-auth/'.format(self.koku_host, self.koku_port)
+        data = {'username': self.admin_user, 'password': self.admin_pass}
+
+        try:
+            response = requests.post(token_auth_url, data=data)
+            if response.status_code != 200:
+                msg = 'Token Auth request return code: {}'.format(str(response.status_code))
+                raise CURAccountsInterfaceError(msg)
+        except requests.exceptions.RequestException as error:
+            raise CURAccountsInterfaceError(str(error))
+
+        response_dict = response.json()
+        token = response_dict.get('token')
+        return token
+
+    def _build_service_admin_auth_header(self):
+        """
+        Construct the service admin Authentication header.
+
+        Args:
+            None
+
+        Returns:
+            ({}) : Authorization header. Example: {'Authorization': Token sdfsdfsd}
+
+        """
+        token = self._get_service_admin_token()
+        token_value = 'Token {}'.format(token)
+        header = {'Authorization': token_value}
+        return header
+
+    def get_accounts_from_source(self):
+        """
+        Retrieve all accounts from the Koku database.
+
+        This will return a list of dicts for the Orchestrator to use to access reports.
+
+        Args:
+            None
+
+        Returns:
+            ([{}]) : A list of dicts
+
+        """
+        providers_url = 'http://{}:{}/api/v1/providers/'.format(self.koku_host, self.koku_port)
+
+        try:
+            response = requests.get(providers_url, headers=self._build_service_admin_auth_header())
```

Comment:
We should check the response here as you might blow up in the next section if its an error that comes back as HTML and you request the `.json()`.

Would it make sense to provide a helper method for this kind of handling?

Reply:
Yeah, I'll make some changes to validate the response before `json()`